src/main/java/fi/metatavu/metamind/server/rest/StoriesApiImpl.java

```diff
@@ -461,45 +462,46 @@ public Response findVariable(UUID storyId, UUID variableId) {
 
   @Override
   public Response listIntents(UUID storyId) {
-    // TODO: Permission check
-    
     fi.metatavu.metamind.persistence.models.Story story = storyController.findStoryById(storyId);
     if (story == null) {
       return createBadRequest(String.format("Story %s not found", storyId)); 
     }
     
+    List<fi.metatavu.metamind.persistence.models.Intent> intents = storyController.listIntentsByStory(story);
+    if (intents.isEmpty()) {
+      return createNotFound(String.format("Intents %s not found", storyId)); 
```

Comment:
Empty list of intents is perfectly normal situation, return 200 with empty list instead of 404